app/src/main/java/com/github/libretube/ui/fragments/PlaylistFragment.kt

```diff
@@ -165,20 +168,35 @@ class PlaylistFragment : BaseFragment() {
                 }
 
                 playlistAdapter = PlaylistAdapter(
-                    response.relatedStreams.orEmpty().toMutableList(),
+                    playlistFeed,
                     playlistId!!,
                     playlistType
                 )
 
                 // listen for playlist items to become deleted
                 playlistAdapter!!.registerAdapterDataObserver(object :
                         RecyclerView.AdapterDataObserver() {
-                        override fun onChanged() {
-                            binding.playlistInfo.text =
-                                binding.playlistInfo.text.split(TextUtils.SEPARATOR).first() + TextUtils.SEPARATOR + getString(
+                        override fun onItemRangeRemoved(positionStart: Int, itemCount: Int) {
+                            if (positionStart == 0) {
+                                ImageHelper.loadImage(
+                                    playlistFeed.firstOrNull()?.thumbnail ?: "",
+                                    binding.thumbnail
+                                )
+                            }
+
+                            val info = binding.playlistInfo.text.split(TextUtils.SEPARATOR)
+                            val countInfo = getString(
```

Comment:
I'd prefer that to be used directly instead of assigning it to a var that's only used once.